metaci/testresults/models.py

```diff
@@ -274,3 +255,300 @@ class TestResultAsset(models.Model):
+
+    @classmethod
+    def _get_queryset_for_dates(cls, start_date, end_date, **values):
+        """Get a queryset for a range of dates."""
+        metrics = cls.metrics()
+
+        buildflows = build_models.BuildFlow.objects.filter(
+            time_end__date__gte=start_date, time_end__date__lte=end_date
+        )
+
+        method_contexts = (
+            TestResult.objects.filter(
+                build_flow_id__in=buildflows,
+                build_flow__build__planrepo_id__isnull=False,
+            )
+            .values(
+                "method_id",
+                rel_repo_id=F("build_flow__build__repo"),
+                rel_branch_id=F("build_flow__build__branch"),
+                rel_plan_id=F("build_flow__build__plan"),
+                rel_planrepo_id=F("build_flow__build__planrepo"),
+                **values,
+            )
+            .annotate(
+                agg_count=Count("method__name"),
+                agg_duration_average=metrics["duration_average"],
+                agg_duration_slow=metrics["duration_slow"],
+                agg_duration_fast=metrics["duration_fast"],
+                agg_cpu_usage_average=metrics["cpu_usage_average"],
+                agg_cpu_usage_low=metrics["cpu_usage_low"],
+                agg_cpu_usage_high=metrics["cpu_usage_high"],
+                agg_failures=metrics["failures"],
+                agg_assertion_failures=metrics["assertion_failures"],
+                agg_DML_failures=metrics["DML_failures"],
+                agg_other_failures=metrics["other_failures"],
+            )
+        )
+        return method_contexts
+
+
+class TestResultPerfWeeklySummaryQuerySet(models.QuerySet):
+    def for_user(self, user, perms=None):
+        if user.is_superuser:
+            return self
+        if perms is None:
+            perms = "plan.view_builds"
+        PlanRepository = apps.get_model("plan.PlanRepository")
+        return self.filter(
+            rel_planrepo__in=PlanRepository.objects.for_user(user, perms)
+        )
+
+
+class TestResultPerfWeeklySummary(TestResultPerfSummaryBase):
+    """Weekly summary table/model"""
+
+    logger = logging.getLogger(__name__)
+    out_hdlr = logging.StreamHandler(sys.stdout)
+    out_hdlr.setLevel(logging.INFO)
+    logger.addHandler(out_hdlr)
+    logger.setLevel(logging.INFO)
+
+    class Meta:
+        verbose_name = "Test Results Weekly Performance Summary"
+        verbose_name_plural = "Test Results Weekly Performance Summaries"
+        db_table = "testresult_weekly_perfsummary"
+        unique_together = ("rel_repo", "rel_branch", "rel_plan", "method", "week_start")
+        indexes = [
+            models.Index(fields=unique_together, name="testresult_weekly_lookup")
+        ]
+
+    week_start = models.DateField(null=False, blank=False)
+    objects = TestResultPerfWeeklySummaryQuerySet.as_manager()
+
+    @classmethod
+    def _parse_start_and_date_dates(cls, start_string, end_string):
```

Comment:
docstring

Reply:
Part of 534c3404ab448173f73b8876f3d361b2642767bc